ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
 #include <catch2/catch.hpp>
 
 #include "gcp_connection.h"
+#include <typeinfo>
 
-TEST_CASE("Testing class gcpConnection", "gcp-connection") {}
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+namespace test_defaults {
+static std::string bucket_name("unit_testing_gcp");
+
+/*
+ * Objects with the prefex pattern "gcptest/*" are deleted after a certain period of time according
+ * to the lifecycle rule on the gcp bucket. Should you wish to make any changes to the prefix
+ * pattern or lifecycle of the object, please speak to the release manager.
+ */
+static std::string obj_prefix("gcptest"); // To be concatenated with a random string.
+} // namespace test_defaults
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294--".
+static int
+randomize_test_prefix()
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+    std::string new_prefix = "gcptest-unit-";
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    new_prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    new_prefix += '/' + std::to_string(my_random_engine());
+    new_prefix += "--";
+
+    test_defaults::obj_prefix = new_prefix;
+
+    return (0);
+}
+
+// Overrides the defaults with the ones specific for this test instance.
+static int
+setup_test_defaults()
```

Comment:
This was added to support other set up feature that we might need in the future, since we only need prefix I removed this function in 082471155d1c61f4100269002a0ab905665652f4